add-on/panels/js/panel.jsx

```diff
@@ -1109,17 +1117,102 @@ loop.panel = (function(_, mozL10n) {
     }
   });
 
+  var RenameRoomView = React.createClass({
+    mixins: [sharedMixins.WindowCloseMixin],
+
+    propTypes: {
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      roomName: React.PropTypes.string.isRequired,
+      roomToken: React.PropTypes.string.isRequired
+    },
+
+    getInitialState: function() {
+      return { focused: false };
+    },
+
+    componentDidMount: function() {
+      this.getDOMNode().querySelector("input").focus();
+    },
+
+    handleBlur: function() {
+      this.setState({ focused: false });
+    },
+
+    handleFocus: function() {
+      this.setState({ focused: true });
+      this.getDOMNode().querySelector("input").select();
+    },
+
+    handleKeyDown: function(event) {
+      if (event.which === 13) {
+        this.handleNameChange();
+      }
+    },
+
+    handleNameChange: function() {
+      let token = this.props.roomToken,
+          name = this.getDOMNode().querySelector("input").value || "";
+      this.props.dispatcher.dispatch(
```

Comment:
According to https://bugzilla.mozilla.org/show_bug.cgi?id=1248568#c2 if the user clicks ok without changing the suggested room name then we should be leaving it as dynamic, i.e. not set the room name - I think you need to address that here.